mdtoc.go

```diff
@@ -1,7 +1,71 @@
 package mdtoc
 
-import "io"
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+)
 
-func Generate(input io.Reader, output io.Writer) {
-	io.Copy(output, input)
+const headerFormat = "- [%s](#%s)"
+
+func writeHeader(output io.Writer, level int, header string) {
+	// TODO: Handle special characters on header
+	line := fmt.Sprintf(headerFormat, header, strings.ToLower(header))
+	// TODO: Handle output writing errors
+	// TODO: Handle when header has trailing spaces
+	output.Write([]byte(line + "\n"))
+}
+
+func parseHeader(parsed []string) (int, string) {
+	var level int
+	for parsed[level] == "#" {
+		level += 1
+	}
+	return level, strings.Join(parsed[level:], "")
+}
+
+func Generate(input io.Reader, output io.Writer) error {
+	headerStart := []byte("<!-- mdtocstart -->")
+	tocHeader := []byte("# Table of Contents")
+	headerEnd := []byte("<!-- mdtocend -->")
+	scanner := bufio.NewScanner(input)
+	var original bytes.Buffer
+	var wroteHeader bool
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		_, err := original.Write([]byte(line + "\n"))
+		if err != nil {
+			return err
+		}
+		parsed := strings.Split(line, "#")
```

Comment:
I appreciate the feedback, but to be honest I'm failing to find using string.Split to be obscure and looping manually byte per byte to be better. Also, not sure if this works properly if there is special characters on the string (like a char with 16 bits that the first most significative byte matches exactly with the ASCII/utf-8 byte for #).

The trimming makes sense, I'm not covering a lot of corner cases yet :+1: